File: backend/app/schemas/audit.py
"""
审计日志 Pydantic Schema
"""
from datetime import datetime

from pydantic import BaseModel, Field


class AuditLogRead(BaseModel):
    model_config = {"from_attributes": True}

    id: int
    timestamp: datetime
    user_id: int | None
    username: str | None
    user_role: str | None
    action_type: str
    target_type: str | None
    target_id: str | None
    ip_address: str | None
    user_agent: str | None
    details: str | None
    result: str


class AuditLogListResponse(BaseModel):
    items: list[AuditLogRead]
    total: int
    page: int
    page_size: int

File: frontend/src/types/audit.ts
/**
 * 审计日志 TypeScript 类型
 */

export interface AuditLog {
  id: number
  timestamp: string
  user_id: number | null
  username: string | null
  user_role: string | null
  action_type: string
  target_type: string | null
  target_id: string | null
  ip_address: string | null
  user_agent: string | null
  details: string | null
  result: string
}

export interface AuditLogListResponse {
  items: AuditLog[]
  total: number
  page: number
  page_size: number
}

export interface AuditQueryParams {
  page?: number
  page_size?: number
  action_type?: string
  user_id?: number
  target_type?: string
}
